Lidgren.Network/NetUPnP.cs

```diff
@@ -214,139 +222,143 @@ private bool CheckAvailability()
+				return false;
+
+			try
+			{
+				SOAPRequest(m_serviceUrl, m_serviceName,
+				"<u:DeletePortMapping xmlns:u=\"urn:schemas-upnp-org:service:" + m_serviceName + ":1\">" +
+				"<NewRemoteHost>" +
+				"</NewRemoteHost>" +
+				"<NewExternalPort>" + externalPort + "</NewExternalPort>" +
+				"<NewProtocol>" + proto + "</NewProtocol>" +
+				"</u:DeletePortMapping>", "DeletePortMapping");
+				return true;
+			}
+			catch (Exception ex)
+			{
+				m_peer.LogWarning("UPnP delete forwarding rule failed: " + ex.Message);
+				return false;
+			}
+		}
+
+		/// <summary>
+		/// Retrieve the extern ip using UPnP
+		/// </summary>
+		public IPAddress? GetExternalIP()
 		{
 			if (!CheckAvailability())
 				return null;
 
 			return GetExternalIP(m_serviceUrl, m_serviceName);
 		}
 
-        private IPAddress GetExternalIP(string url, string name)
-        {
-	        try
-	        {
-		        XmlDocument xdoc = SOAPRequest(url, name, "<u:GetExternalIPAddress xmlns:u=\"urn:schemas-upnp-org:service:" + name + ":1\">" +
-		                                                  "</u:GetExternalIPAddress>", "GetExternalIPAddress");
-		        XmlNamespaceManager nsMgr = new XmlNamespaceManager(xdoc.NameTable);
-		        nsMgr.AddNamespace("tns", "urn:schemas-upnp-org:device-1-0");
-		        string IP = xdoc.SelectSingleNode("//NewExternalIPAddress/text()", nsMgr).Value;
-		        return IPAddress.Parse(IP);
-	        }
-	        catch (Exception ex)
-	        {
-		        m_peer.LogWarning("Failed to get external IP: " + ex.Message);
-		        return null;
-	        }
-        }
-
-        private string GetConnectionStatus(string url, string name)
-        {
-	        try
-	        {
-		        XmlDocument xdoc = SOAPRequest(
-			        url, name,
-			        $"<u:GetStatusInfo xmlns:u=\"urn:schemas-upnp-org:service:{name}:1\" />",
-			        "GetStatusInfo");
-		        XmlNamespaceManager nsMgr = new XmlNamespaceManager(xdoc.NameTable);
-		        nsMgr.AddNamespace("tns", "urn:schemas-upnp-org:device-1-0");
-		        return xdoc.SelectSingleNode("//NewConnectionStatus/text()", nsMgr).Value;
-	        }
-	        catch (Exception ex)
-	        {
-		        m_peer.LogWarning("Failed to get connection status: " + ex.Message);
-		        return null;
-	        }
-        } 
-
-		private XmlDocument SOAPRequest(string url, string serviceName, string soap, string function)
+		private IPAddress? GetExternalIP(string? url, string name)
+		{
+			try
+			{
+				XmlDocument xdoc = SOAPRequest(url, name, "<u:GetExternalIPAddress xmlns:u=\"urn:schemas-upnp-org:service:" + name + ":1\">" +
+														  "</u:GetExternalIPAddress>", "GetExternalIPAddress");
+				XmlNamespaceManager nsMgr = new XmlNamespaceManager(xdoc.NameTable);
+				nsMgr.AddNamespace("tns", "urn:schemas-upnp-org:device-1-0");
+				string? IP = xdoc.SelectSingleNode("//NewExternalIPAddress/text()", nsMgr)?.Value;
+
+				if (IP == null)
+				{
+					return null;
+				}
+

```

Comment:
Same thing as before. This turns an exception into a silent failure. Logging would be best.